src/dispatcher.ts

```diff
@@ -150,9 +150,11 @@ export class Dispatcher {
       const requiredHash = file.hash;
       let worker = await this._obtainWorker();
       while (!this._isStopped && worker.hash && worker.hash !== requiredHash) {
-        this._restartWorker(worker);
+        worker.stop();
```

Comment:
To make sure I understand, for 3 workers, 3 tests with hash "A" and 3 tests with hash "B", we create 3 workers for the first 3 tests. Then we wait for them to become available. We get three workers back and we schedule 'stop' for them. Then we start obtaining the new one. There are no free workers and `this._workers.size` is still 3, so we become a claimer. But who is going to create a new worker?

Reply:
Fixed!